### PR TITLE
validate user has accepted terms on the server

### PR DESCRIPTION
**Pull request by tavvy**
This pr fixes a bunch of underlying bugs and errors and also surface errors within the error message dialog when using test apis for easier debugging via nightwatch videos etc.

Furthermore we are now validating that the user has accepted the T&C's serverside.

- [x] need to throw a nicer error for cc/dd in logs
- [x] show in error message box

---

Because this is a revenue-generating app, we want to ensure changes will not affect its basic functionality. So for external teams wanting to submit pull requests, please do the following in addition to the regular pull request procedure:

- [ ] Make sure your branch passes CI checks before merging
- [ ] Add unit/smoke/regression tests for new functionality as needed. If unsure, ask in #ft-next-signup

### CODE REVIEW

File: client/routes/signup-form/hosted-form.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ export default class HostedForm {
 
 	static onSubmissionSucceeding (callback) {
 		HostedForm.onSubmissionResult((result)=> {
-			if(result.success) {
+			if(result.success && result.success === 'true') {
```
Comment (tavvy) [inline client/routes/signup-form/hosted-form.js]: This should fix some errors within the app.

File: client/routes/signup-form/hosted-form.js [hunk 2]
```diff
@@ -30,12 +30,12 @@ export default class HostedForm {
...
 
-			if (result && result.code && result.message && result.code === 'HostedPageFieldValidationError') {
-				// store 3rd party Zuora server errors
+			// store 3rd party Zuora server errors
+			if (result && result.code && result.code === 'HostedPageFieldValidationError' && result.message) {
 				Zuora.serverError = result.message || false;
 			}
 
-			if(!result.success) {
+			if(!result.success || result.success === 'false') {
```
Comment (tavvy) [inline client/routes/signup-form/hosted-form.js]: As should this

File: server/policies/error-handling.js [hunk 3]
```diff
@@ -4,7 +4,8 @@ import MaskLogger from '@financial-times/n-mask-logger';
 
 const logger = new MaskLogger(['email', 'password']);
 
-export default function (err, req, res) {
+// !NOTE: all four params are REQUIRED
+export default function (err, req, res, next) {
```
Comment (tavvy) [inline server/policies/error-handling.js]: Another bug fix

File: server/routes/api/subscription/post.js [hunk 4]
```diff
@@ -112,18 +113,20 @@ export default function (req, res, next) {
...
 			}
-			if(the.mandate.signing.failed) {
+			else if(the.subscription.creation.failed) {
 				er.render503(error);
+				silentlySubmitSubscriptionCreationFailureTrackingEvent();
+			}
+			else {
+				er.render500(error);
 			}
-			er.render500(error);
```
Comment (tavvy) [inline server/routes/api/subscription/post.js]: I believe this is also the cause of a lot of issues. We could have been rendering two errors (one from the `if` statements above, **and** the `500` error.

**Comment by ironsidevsquincy** [comment]
👍 

Once this is out we could try turning on the browsers we turned off when testing

**Pull request merged**